test/test_sdo.py

```diff
@@ -170,6 +170,314 @@ def test_block_upload(self):
+            (RX, b"\x01\x74\x68\x65\x20\x63\x72\x61"),
+            (RX, b"\x02\x7a\x79\x20\x66\x6f\x78\x20"),
+            (RX, b"\x03\x6a\x75\x6d\x70\x73\x20\x6f"),
+            (RX, b"\x04\x76\x65\x72\x20\x74\x68\x65"),
+            (RX, b"\x05\x20\x6c\x61\x7a\x79\x20\x64"),
+            (RX, b"\x06\x6f\x67\x0a\x74\x68\x65\x20"),
+            (RX, b"\x07\x63\x72\x61\x7a\x79\x20\x66"),
+            (RX, b"\x08\x6f\x78\x20\x6a\x75\x6d\x70"),
+            (RX, b"\x09\x73\x20\x6f\x76\x65\x72\x20"),
+            (RX, b"\x0a\x74\x68\x65\x20\x6c\x61\x7a"),
+            (RX, b"\x0b\x79\x20\x64\x6f\x67\x0a\x74"),
+            (RX, b"\x0c\x68\x65\x20\x63\x72\x61\x7a"),
+            (RX, b"\x0d\x79\x20\x66\x6f\x78\x20\x6a"),
+            (RX, b"\x0e\x75\x6d\x70\x73\x20\x6f\x76"),
+            (RX, b"\x0f\x65\x72\x20\x74\x68\x65\x20"),
+            (RX, b"\x10\x6c\x61\x7a\x79\x20\x64\x6f"),
+            (RX, b"\x11\x67\x0a\x74\x68\x65\x20\x63"),
+            (RX, b"\x12\x72\x61\x7a\x79\x20\x66\x6f"),
+            (RX, b"\x13\x78\x20\x6a\x75\x6d\x70\x73"),
+            (RX, b"\x14\x20\x6f\x76\x65\x72\x20\x74"),
+            (RX, b"\x15\x68\x65\x20\x6c\x61\x7a\x79"),
+            (RX, b"\x16\x20\x64\x6f\x67\x0a\x74\x68"),
+            (RX, b"\x17\x65\x20\x63\x72\x61\x7a\x79"),
+            (RX, b"\x18\x20\x66\x6f\x78\x20\x6a\x75"),
+            (RX, b"\x19\x6d\x70\x73\x20\x6f\x76\x65"),
+            (RX, b"\x1a\x72\x20\x74\x68\x65\x20\x6c"),
+            (RX, b"\x1b\x61\x7a\x79\x20\x64\x6f\x67"),
+            (RX, b"\x1c\x0a\x74\x68\x65\x20\x63\x72"),
+            (RX, b"\x1d\x61\x7a\x79\x20\x66\x6f\x78"),
+            (RX, b"\x1e\x20\x6a\x75\x6d\x70\x73\x20"),
+            (RX, b"\x1f\x6f\x76\x65\x72\x20\x74\x68"),
+            (RX, b"\x20\x65\x20\x6c\x61\x7a\x79\x20"),
+            (RX, b"\x21\x64\x6f\x67\x0a\x74\x68\x65"),
+            (RX, b"\x22\x20\x63\x72\x61\x7a\x79\x20"),
+            (RX, b"\x23\x66\x6f\x78\x20\x6a\x75\x6d"),
+            (RX, b"\x24\x70\x73\x20\x6f\x76\x65\x72"),
+            (RX, b"\x25\x20\x74\x68\x65\x20\x6c\x61"),
+            (RX, b"\x26\x7a\x79\x20\x64\x6f\x67\x0a"),
+            (RX, b"\x27\x74\x68\x65\x20\x63\x72\x61"),
+            (RX, b"\x28\x7a\x79\x20\x66\x6f\x78\x20"),
+            (RX, b"\x29\x6a\x75\x6d\x70\x73\x20\x6f"),
+            (RX, b"\x2a\x76\x65\x72\x20\x74\x68\x65"),
+            (RX, b"\x2b\x20\x6c\x61\x7a\x79\x20\x64"),
+            (RX, b"\x2c\x6f\x67\x0a\x74\x68\x65\x20"),
+            (RX, b"\x2d\x63\x72\x61\x7a\x79\x20\x66"),
+            (RX, b"\x2e\x6f\x78\x20\x6a\x75\x6d\x70"),
+            (RX, b"\x2f\x73\x20\x6f\x76\x65\x72\x20"),
+            (RX, b"\x30\x74\x68\x65\x20\x6c\x61\x7a"),
+            (RX, b"\x31\x79\x20\x64\x6f\x67\x0a\x74"),
+            (RX, b"\x32\x68\x65\x20\x63\x72\x61\x7a"),
+            (RX, b"\x34\x79\x20\x66\x6f\x78\x20\x6a"),
+            (RX, b"\x34\x75\x6d\x70\x73\x20\x6f\x76"),  # --> Wrong seqno (x34 instead of x33)
```

Comment:
Are these lines intentionally using the same sequence number?  From the comment I'd assume the previous segment should have been missing, not misunderstood.